### PR TITLE
Add New package feh and supporting packages

### PR DESCRIPTION
**Pull request by TheQueasle**
### Package

* **Name**: feh
* **Description**: feh is an X11 image viewer aimed mostly at console users. Unlike most other viewers, it does not have a fancy GUI, but simply displays images. It is controlled via commandline arguments and configurable key/mouse actions.

### Supporting Packages
* **imlib2**: Library that does image file loading and saving as well as rendering, manipulation, arbitrary polygon support
* **libexif**: A library to parse an EXIF file and read the data from those tags
* **libid3tag**: library for id3 tagging

---
If need be I can resubmit this as 4 separate Pull Requests but as the 3 extra packages are just to support the topic of this PR (feh), I decided to submit one request that includes all 4 packages.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libid3tag/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @run_before('configure')
+    def preclean(self):
+        """
+        Remove compat.c and frametype.c in order to regenerate from gperf
+        sources
+        """
+        rm = which('rm')
+        rm('-v', 'compat.c')
+        rm('-v', 'frametype.c')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libid3tag/package.py]: I would use `os.remove(filename)` to be completely platform agnostic. Spack is planning on supporting Windows in the future.

File: var/spack/repos/builtin/packages/libid3tag/package.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    version('0.15.1b', 'e5808ad997ba32c498803822078748c3')
+
+    depends_on('zlib')
+    depends_on('gperf')
+
+    patch('10_utf16.diff')
+    patch('11_unknown_encoding.dif')
+    patch('CVE-2008-2109.patch', level=0)
+    patch('libid3tag-gperf.patch', when="^gperf@3.1:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libid3tag/package.py]: Could you add a link to where these patches came from? Or submit them upstream if you wrote any of them yourself?

File: var/spack/repos/builtin/packages/libid3tag/package.py (hunk 1 above)
Comment (TheQueasle) [reply var/spack/repos/builtin/packages/libid3tag/package.py]: Will do.  Thanks for the suggestion!

I actually was looking for a better way to do that but didn't look hard enough it seems :smile:

File: var/spack/repos/builtin/packages/libid3tag/package.py (hunk 2 above)
Comment (TheQueasle) [reply var/spack/repos/builtin/packages/libid3tag/package.py]: Yes - I'll add the source for these files.  Where should I place the links?  Directly in the package file as a comment?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/libid3tag/package.py]: I either put them at the top of the patch files themselves, or in the `package.py` above each patch. Either is fine, just depends on how cluttered you want things to get.

File: var/spack/repos/builtin/packages/libid3tag/package.py (hunk 1 above)
Comment (TheQueasle) [reply var/spack/repos/builtin/packages/libid3tag/package.py]: Swapped `rm` for `os.remove`

File: var/spack/repos/builtin/packages/libid3tag/package.py (hunk 2 above)
Comment (TheQueasle) [reply var/spack/repos/builtin/packages/libid3tag/package.py]: Added URLs to the sources for those patches

**Pull request merged**